### DISCUSSION TITLE
Jop shop problem with priority

### DISCUSSION THREAD

**Discussion by Bestbbb** [google/or-tools#3574] [CP-SAT questions] [answered]
If every task has a priority, the requirement is to ask in the same day, higher priority task finished more quantity than lower priority task,how can I achieve it? Anyone has a convinient method？

**Comment by lperron** [comment, marked as answer]
Tardiness cost in jobshop_sat.cc

**Reply by Bestbbb** [reply]
Have it been achieved in other languages?

**Comment by lperron** [comment]
The API is the same across languages. Just adapt it.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by Bestbbb** [reply]
Thanks,I will try it.

**Reply by Bestbbb** [reply]
Sorry to bother you, there is indeed a Tardiness cost in jobshop_sat.cc, but I checked the documentation, including python and java, and found no similar api, or an api similar to this cost. If you can, please help me find a similar cost API?

**Comment by lperron** [comment]
You are mixing two things:

- how to implement tardiness: that is `tardiness_var == max(0, end_date - due_date)`. This is easy to code in all languages.
- where it is implemented: jobshop_sat.cc as this is the only generic jobshop code

from there, you should be able to adapt the C++ code and add it to your example in Python/Java/.NET

**Reply by Bestbbb** [reply]
OK,I see, thank you so much.Have a nice day (*￣︶￣)
